apps/etterlatte-behandling/src/main/kotlin/oppgave/OppgaveServiceImpl.kt

```diff
@@ -0,0 +1,41 @@
+package no.nav.etterlatte.oppgave
+
+import no.nav.etterlatte.Saksbehandler
+import no.nav.etterlatte.inTransaction
+import no.nav.etterlatte.libs.common.behandling.BehandlingStatus
+import no.nav.etterlatte.oppgave.domain.Oppgave
+
+interface OppgaveService {
+
+    fun finnOppgaverForBruker(bruker: Saksbehandler): List<Oppgave>
+}
+
+class OppgaveServiceImpl(private val oppgaveDao: OppgaveDao) : OppgaveService {
+
+    private fun finnAktuelleRoller(bruker: Saksbehandler): List<Rolle> =
+        listOfNotNull(
+            Rolle.SAKSBEHANDLER.takeIf { bruker.harRolleSaksbehandler() },
+            Rolle.ATTESTANT.takeIf { bruker.harRolleAttestant() }
+        )
+
+    private fun aktuelleStatuserForRolleTilSaksbehandler(roller: List<Rolle>) = roller.flatMap {
+        when (it) {
+            Rolle.SAKSBEHANDLER -> BehandlingStatus.kanEndres()
+
+            Rolle.ATTESTANT -> listOf(BehandlingStatus.FATTET_VEDTAK)
+        }
+    }.distinct()
+
+    override fun finnOppgaverForBruker(bruker: Saksbehandler): List<Oppgave> {
+        val rollerSomBrukerHar = finnAktuelleRoller(bruker)
+        val aktuelleStatuserForRoller = aktuelleStatuserForRolleTilSaksbehandler(rollerSomBrukerHar)
+
+        val alleOppgaver = inTransaction {
+            listOf(
+                oppgaveDao.finnOppgaverMedStatuser(aktuelleStatuserForRoller),
```

Comment:
Kan vi gjøre oppslagene samtidig? :D

Reply:
Skal ta en titt på det, men måten jeg ser for meg å gjøre dette på blir enten å flytte oppdelingen i to spørringer til OppgaveDao (men de blir ikke samtidige der heller), eller å smelte dette sammen til en mer kompleks SQL-spørring (og her er jeg ikke helt sikker på hva vi vinner i så fall). Har du noen forslag / ideer til hvordan dette evt. kan gjøres?

Reply:
Hver sin spørring i samme transaction er fint det, var mer om de kunne gjøres parallelt :)